design/ceph/object/store.md

```diff
@@ -440,3 +440,284 @@ For external CephCephObjectStores (i.e., when `spec.gateway.externalRgwEndpoints
+> `data_extra_pool` is for data that cannot use erasure coding. For example, multi-part uploads allow uploading a large object such as a movie in multiple parts. These parts must first be stored without erasure coding. So if `data_pool` is **not** erasure coded, then there is not need for `data_extra_pool`.
+
+```yaml
+# Default Data pool
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: default.rgw.buckets.data
+  namespace: rook-ceph
+spec:
+  ...
+---
+# Default index pool
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: default.rgw.buckets.index
+  namespace: rook-ceph
+spec:
+  ...
+# Bucket Data pool for hot storage
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: hot.rgw.buckets.data
+  namespace: rook-ceph
+spec:
+  ...
+---
+# Bucket index pool for hot storage
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: hot.rgw.buckets.index
+  namespace: rook-ceph
+spec:
+  ...
+---
+# Bucket Data pool for cold storage
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: cold.rgw.buckets.data
+  namespace: rook-ceph
+spec:
+  ...
+---
+# Bucket index pool for cold storage
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: cold.rgw.buckets.index
+  namespace: rook-ceph
+spec:
+  ...
+```
+
+If needed, create additional pool to use it in custom StorageClass, for example `GLACIER` and `REDUCED_REDUNDANCY`:
+> [!NOTE]
+> Ceph allows arbitrary name for StorageClasses, however some clients/libs insist on AWS names so it is recommended to use one of the [valid x-amz-storage-class values](https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutObject.html#API_PutObject_RequestSyntax) for better compatibility:
+>
+> `STANDARD | REDUCED_REDUNDANCY | STANDARD_IA | ONEZONE_IA | INTELLIGENT_TIERING | GLACIER | DEEP_ARCHIVE | OUTPOSTS | GLACIER_IR | SNOW | EXPRESS_ONEZONE`
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: glacier.rgw.buckets.data
+  namespace: rook-ceph
+spec:
+  ...
+---
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: reduced.rgw.buckets.data
+  namespace: rook-ceph
+spec:
+  ...
+```
+
+Then update `CephObjectStore` CRD  `sharedPools` section with target placements and StorageClasses definitions:
+
+```yaml
+spec:
+  # [OPTIONAL] - sharedPools is mutually exclusive with spec.metadataPool and spec.dataPool
+  # It is the place to tell RGW which pools can be used to store buckets and objects data and metadata.
+  # All pools under this section should be created beforehand.
+  # Rook will append rados namespace to each of the pools, so it is safe to use the same pool across multiple RGW instances.
+  # It is also safe to use the same pool in different target placements.
+  sharedPools:
+    #  [REQUIRED] - metadata pool. Will be used by default to store bucket index
+    metadataPoolName: default.rgw.buckets.index # results to default.rgw.buckets.index:<rgw_name>.index
```

Comment:
Something I missed the other day, I think `rgw_name` here should be intended to mean CephObjectStore.name, for clarity. it's okay for multiple RGWs to share the same pool, but they should be part of the same CephObjectStore (or CephObjectZone) I think.

Reply:
fixed